src/clients/github.js

```diff
@@ -0,0 +1,276 @@
+import get from 'lodash/get';
+import GitHub from 'github-api';
+import isEmpty from 'lodash/isEmpty';
+import trim from 'lodash/trim';
+import performWithRetries from '../util/performWithRetries';
+import generatePreview from '../util/generatePreview';
+
+const _anonymousGithub = new GitHub({});
+const COMMIT_MESSAGE = 'Created using Popcode: https://popcode.org';
+const MASTER = 'master';
+const GH_PAGES = 'gh-pages';
+const NETWORK_ERROR = 'Network Error';
+const parser = new DOMParser();
+
+export function EmptyGistError(message) {
+  this.name = 'EmptyGistError';
+  this.message = message;
+  this.stack = new Error().stack;
+}
+EmptyGistError.prototype = Object.create(Error.prototype);
+
+function getPageTitle(preview) {
+  const parsedDocument = parser.parseFromString(preview, 'text/html');
+
+  const titleWithoutPunctuationAndWhitespace = parsedDocument.title.
+    replace(/[^\w\s]|_/g, '').
+    replace(/\W/g, '-');
+
+  return titleWithoutPunctuationAndWhitespace;
+}
+
+export async function createRepoFromProject(project, user) {
+  const github = clientForUser(user);
+  const preview = generatePreview(project);
+  const title = getPageTitle(preview);
+
+  const REPO_NAME = `${title}-${Date.now()}`;
+
+  const {data} =
+    await performWithRetryNetworkErrors(
+      () => github.getUser().createRepo({name: REPO_NAME, auto_init: true}),
```

Comment:
Doesn't seem to work.

First it does a GET for `https://api.github.com/repos/alexpelan/Page-Title-1499912981497/contents/index.html?ref=gh-pages`

response: 

    {
      "message": "No commit found for the ref gh-pages",
      "documentation_url": "https://developer.github.com/v3/repos/contents/"
    }

Then a PUT for `https://api.github.com/repos/alexpelan/Page-Title-1499912981497/contents/index.html`

response:

    {
       "message": "Branch gh-pages not found",
       "documentation_url": "https://developer.github.com/v3/repos/contents/"
    }